controller/cmd/public-api/main.go

```diff
@@ -56,13 +58,21 @@ func main() {
 		log.Fatal(err.Error())
 	}
 
+	globalConfig, err := config.Global(pkgK8s.MountPathGlobalConfig)
+	clusterDomain := globalConfig.GetClusterDomain()
+	if err != nil || clusterDomain == "" {
```

Comment:
I think we should handle the two conditionals separately here, if the intent is to fall back to `cluster.local` when `clusterDomain` is empty. How about:

```go
globalConfig, err := config.Global(pkgK8s.MountPathGlobalConfig)
if err != nil {
  log.Fatal(err)
}
clusterDomain := globalConfig.GetClusterDomain()
if clusterDomain == "" {
  clusterDomain = "cluster.local"
}
log.Info("Using cluster domain: ", clusterDomain)
```

Reply:
I like it as long as we can see the cluster domain somewhere in the startup process. Thanks for that!